src/olympia/addons/models.py

```diff
@@ -706,29 +706,30 @@ def find_latest_public_listed_version(self):
         except (IndexError, Version.DoesNotExist):
             return None
 
-    def find_latest_version(self, channel, ignore=None):
-        """Retrieve the latest non-disabled version of an add-on for the
-        specified channel.  If channel is None either channel is returned.
+    def find_latest_version(
+            self, channel, exclude=(amo.STATUS_DISABLED, amo.STATUS_BETA)):
+        """Retrieve the latest version of an add-on for the specified channel.
+
+        If channel is None either channel is returned.
+
+        Keyword arguments:
+        exclude -- exclude versions for which all files have one
+                   of those statuses (default STATUS_DISABLED, STATUS_BETA)."""
 
-        Accepts an optional ignore argument to ignore a specific version."""
         # If the add-on is deleted or hasn't been saved yet, it should not
         # have a latest version.
         if not self.id or self.status == amo.STATUS_DELETED:
             return None
-        # We can't use .exclude(files__status=STATUS_DISABLED) because this
-        # excludes a version if any of the files are the disabled and there may
-        # be files we do want to include.  Having a single beta file /does/
-        # mean we want the whole version disqualified though.
-        statuses_without_disabled = (
-            set(amo.STATUS_CHOICES_FILE.keys()) -
-            {amo.STATUS_DISABLED, amo.STATUS_BETA})
+        # We can't use .exclude(files__status=excluded_statuses) because that
+        # would exclude a version if *any* of its files match but if there is
+        # only one file that doesn't have one of the excluded statuses it
+        # should be enough for that version to be considered.
+        statuses_no_disabled_or_beta = (
+            set(amo.STATUS_CHOICES_FILE.keys()) - set(exclude))
         try:
             latest_qs = (
                 Version.objects.filter(addon=self)
-                       .exclude(files__status=amo.STATUS_BETA)
-                       .filter(files__status__in=statuses_without_disabled))
-            if ignore is not None:
-                latest_qs = latest_qs.exclude(pk=ignore.pk)
+                       .filter(files__status__in=statuses_no_disabled_or_beta))
```

Comment:
It was intended, because I thought this shouldn't happen: you can't have different version numbers in files belonging to the same version, so if one of the files is a beta, the whole version should be, right ?